scripts/extensions/pool_update.apply

```diff
@@ -118,7 +117,11 @@ if __name__ == '__main__':
             print(failure_message(PATCH_APPLY_FAILED,
                                   ['Apply failed: %s' % e]))
         finally:
-            session.xenapi.pool_update.detach(update)
+            try:
+                session.xenapi.pool_update.detach(update)
+            except Exception as e:
+                print(failure_message(PATCH_APPLY_FAILED,
+                    ['Apply failed: %s' % e]))
```

Comment:
Thanks @thomassa. Here `PATCH_APPLY_FAILED` is returned as an error code so that client like XenCenter can handle.